### PR TITLE
argu, fixed

### PR DESCRIPTION
**Pull request by eKIRILLmm**

### CODE REVIEW

File: src/YaccConstructor/YaccConstructor.fsproj [hunk 1]
```diff
@@ -68,6 +68,9 @@
       <Project>{58497973-2ae7-4e48-a6fc-9ed7a5c6cce6}</Project>
       <Private>True</Private>
     </ProjectReference>
+    <Reference Include="Argu">
+      <HintPath>..\..\..\..\projects\yaccconstructor — копия\packages\Argu.3.2.0\lib\net40\Argu.dll</HintPath>
```
Comment (auduchinok) [inline src/YaccConstructor/YaccConstructor.fsproj]: What does this path stand for?

File: src/Constraints/Constraints.fsproj [hunk 2]
```diff
@@ -146,4 +109,46 @@
...
+    <Reference Include="mscorlib" />
+    <Reference Include="System" />
+    <Reference Include="System.Core" />
+    <Reference Include="System.Numerics" />
+    <ProjectReference Include="..\Common\Common.fsproj">
+      <Name>Common</Name>
+      <Project>{10ea63ab-6f4e-4925-b0a3-4f9bf8e17a08}</Project>
+      <Private>True</Private>
+    </ProjectReference>
+  </ItemGroup>
```
Comment (auduchinok) [inline src/Constraints/Constraints.fsproj]: Is it necessary to move this `ItemGroup`?

File: src/YaccConstructor/Program.fs [hunk 3]
```diff
@@ -44,6 +44,32 @@ let log (e:System.Exception) msg =
     "\nMessage:\n  " + msg
     |> printfn "%s"
 
+open Argu
+
+type CLIArguments =
+    | [<Inherit; AltCommandLine("-f","-frontend")>] F of feName:string option
```
Comment (auduchinok) [inline src/YaccConstructor/Program.fs]: Please use more descriptive constructor names. It's more difficult to get what they are in other places in the code.

File: src/YaccConstructor/Program.fs [hunk 4]
```diff
@@ -88,37 +114,47 @@ let () =
             Some tmpName
         else None
 
+    let generatorNameD = 
+            match !generatorName with
+            |Some x->x
+            |None->""
```
Comment (auduchinok) [inline src/YaccConstructor/Program.fs]: Please format code better.

File: src/YaccConstructor/Program.fs [hunk 5]
```diff
@@ -88,37 +114,47 @@ let () =
...
-         "-D", ArgType.String (fun s -> userDefs := !userDefs @ [s]), "User defined constants for YardFrontend lexer."
-         "-U", ArgType.String (fun s -> userDefs := List.filter ((<>) s) !userDefs), 
-                "Remove previously defined constants for YardFrontend lexer."
-         "-i", ArgType.String (fun s ->
-                                   testFile := System.IO.Path.GetFileName s |> Some
-                                   testsPath := System.IO.Path.GetDirectoryName s |> Some), "Input grammar"         
-         ] |> List.map (fun (shortcut, argtype, description) -> ArgInfo(shortcut, argtype, description))
-    ArgParser.Parse commandLineSpecs
+
+    for c in args.GetResults <@ C @> do
```
Comment (auduchinok) [inline src/YaccConstructor/Program.fs]: This long list of loops and conditions can be replaced with pattern-matching while iterating over all arguments.

File: src/YaccConstructor/Program.fs (hunk 3 above)
Comment (gsvgit) [inline src/YaccConstructor/Program.fs]: Seems that Argu uses constructor name as default key name. So you can move full names from AltCommandLine attribute to constructor name.

File: src/YaccConstructor/Program.fs [hunk 6]
```diff
@@ -44,6 +44,32 @@ let log (e:System.Exception) msg =
     "\nMessage:\n  " + msg
     |> printfn "%s"
 
+open Argu
+
+type CLIArguments =
+    | [<Inherit; AltCommandLine("-f","-frontend")>] F of feName:string option
+    | [<AltCommandLine("-af","-availableFrontends")>] Af
```
Comment (gsvgit) [inline src/YaccConstructor/Program.fs]: [```Unique``` ] (http://fsprojects.github.io/Argu/reference/argu-arguattributes-uniqueattribute.html) attribute should be specified for some args.

File: src/YaccConstructor/Program.fs [hunk 7]
```diff
@@ -88,37 +114,47 @@ let () =
...
-        ["-f", ArgType.String (fun s -> feName := Some s), "Frontend name. Use -af to list available."
-         "-af", ArgType.Unit (printItems "frontends" addinFrontendNames !feName), "Available frontends"
-         "-g", ArgType.String 
-            (fun s -> 
-                match Array.toList (s.Split ' ') with
+    let argv = System.Environment.GetCommandLineArgs()
+    let parser = ArgumentParser.Create<CLIArguments>(errorHandler = ProcessExiter())
+    let args = parser.Parse argv.[1..]
+    feName := args.GetResult (<@ F @>, defaultValue = !feName)
+    let genString= args.GetResult (<@ G @>, defaultValue = generatorNameD)
```
Comment (gsvgit) [inline src/YaccConstructor/Program.fs]: Be careful with spaces.

File: src/YaccConstructor/YaccConstructor.fsproj (hunk 1 above)
Comment (gsvgit) [inline src/YaccConstructor/YaccConstructor.fsproj]: Really strange path

**Comment by gsvgit** [comment]
... and build still failed

**Pull request closed without merging**